### PR TITLE
Fix #267: Delete all releases of an app upon certificate change

### LINKED ISSUE #267: API: changing the certificate for an app should delete all of its releases
Since they can not be validated anymore

### PR DESCRIPTION
**Pull request by janibonnevier**
#267

### CODE REVIEW

File: nextcloudappstore/core/models.py [hunk 1]
```diff
@@ -203,6 +203,17 @@ def _latest(self, releases):
...
             return None
 
+    def save(self, *args, **kwargs):
+        # If the certificate has changed, delete all releases.
+        try:
+            if self.pk is not None:
+                orig = App.objects.get(pk=self.pk)
+                if self.certificate != orig.certificate:
+                    self.releases.all().delete()
+        except self.DoesNotExist:
```
Comment (janibonnevier) [inline nextcloudappstore/core/models.py]: I don't understand where this exception class is.

Comment (BernhardPosselt) [inline nextcloudappstore/core/models.py]: App.DoesNotExist

**Comment by BernhardPosselt** [comment]
:+1:

File: nextcloudappstore/core/models.py (hunk 1 above)
Comment (janibonnevier) [inline nextcloudappstore/core/models.py]: Apparently...

**Pull request merged**